libsolidity/inlineasm/AsmAnalysis.cpp

```diff
@@ -232,8 +232,9 @@ bool AsmAnalyzer::operator()(assembly::FunctionCall const& _funCall)
 		},
 		[&](Scope::Function const& _fun)
 		{
-			arguments = _fun.arguments;
-			returns = _fun.returns;
+			/// TODO: compare types too
```

Comment:
This has to be done in a second pass through the AST because it is possible to call functions that are declared after they are called.